bookkeeper-server/src/test/java/org/apache/bookkeeper/proto/MockBookies.java

```diff
@@ -123,6 +123,47 @@ public ByteBuf readEntry(BookieId bookieId, int flags, long ledgerId, long entry
         return entry;
     }
 
+    public ByteBufList batchReadEntry(BookieId bookieId, int flags, long ledgerId, long startEntryId,
+            int maxCount, long maxSize) throws BKException {
+        MockLedgerData ledger = getBookieData(bookieId).get(ledgerId);
+
+        if (ledger == null) {
+            LOG.warn("[{};L{}] ledger not found", bookieId, ledgerId);
+            throw new BKException.BKNoSuchLedgerExistsException();
+        }
+
+        if ((flags & BookieProtocol.FLAG_DO_FENCING) == BookieProtocol.FLAG_DO_FENCING) {
+            ledger.fence();
+        }
+        //Refer: BatchedReadEntryProcessor.readData
+        ByteBufList data = null;
+        if (maxCount <= 0) {
+            maxCount = Integer.MAX_VALUE;
+        }
+        long frameSize = 24 + 8 + 4;
+        for (long i = startEntryId; i < startEntryId + maxCount; i++) {
+            try {
+                ByteBuf entry = ledger.getEntry(i);
+                frameSize += entry.readableBytes() + 4;
+                if (data == null) {
+                    data = ByteBufList.get(entry);
+                } else {
+                    if (frameSize > maxSize) {
+                        entry.release();
+                        break;
+                    }
+                    data.add(entry);
+                }
+            } catch (Throwable e) {
```

Comment:
catching Throwable is always a code smell, why ? what do you want to catch here ?

Reply:
Make sense, I copy the code from the BatchedReadEntryProcessor, here we don't need catch the exception.